src/kayenta/layout/disableable.tsx

```diff
@@ -32,7 +32,7 @@ function disableable<T extends IDisableable>(Component: React.SFC<T>) {
     const { disabled, disabledBecauseOfState } = props;
 
     // Would use object spread except for weird interaction with TS generics.
-    const otherProps = omit(props, ['disabled', 'disabledBecauseOfState', 'disabledStateKeys', 'dispatch']);
+    const otherProps: T = omit(props, ['disabled', 'disabledBecauseOfState', 'disabledStateKeys', 'dispatch']);
```

Comment:
this is the worst, but the whole `T extends IDisableable` thing is fundamentally borked on anything above 3.2 and I didn't want to couple together a whole big rethinking of how this is typed. It confuses me and I don't like it.